### DISCUSSION TITLE
Extra argument or keyword error spike prime vscode

### DISCUSSION THREAD

**Discussion by Xylar52** [pybricks/support#1050] [Q&A]
I am using spike prime with vscode for now a month and there have been no problems. But today when i installed the last pre-release of the pybricks library and the last pre release of the firmware (beta.pybricks.com), pybricksdev couldn't send the program to the robot because of 
`DEBUG: pybricksdev.compile: missing modules: ['pybricks.hubs', 'pybricks.parameters', 'pybricks.pupdevices', 'pybricks.robotics', 'pybricks.tools']`
The libraries in the error were all the libraries i imported in the code

**Comment by dlech** [comment]
That is just a debug message which is expected. Can you please give the exact steps to reproduce the problem?

**Reply by Xylar52** [reply]
It's simple it's just a basic code and I sent it to the robot with `pybricksdev --debug run ble test_gyro.py` 
But if you think that's normal, I have another problem. At the same time I had an error that said the line 17 had an extra keyword or argument (see line 17 below). The error appeared only after installing the lastest library and firmware, and I didn't touch the code in between.

Code (line 17 is when I initialize drivebase) :

```python
from pybricks.hubs import PrimeHub
from pybricks.pupdevices import Motor, ColorSensor, UltrasonicSensor, ForceSensor
from pybricks.parameters import Button, Color, Direction, Port, Side, Stop, Icon
from pybricks.robotics import DriveBase
from pybricks.tools import wait, StopWatch

#Initialisation des différents senseurs et moteurs et commandes de début de code
hub = PrimeHub()
motorl = Motor(Port.A, Direction.COUNTERCLOCKWISE)
motorr = Motor(Port.B)
motorl.control.limits(acceleration=[2000, 400])
motorr.control.limits(acceleration=[2000, 400])
hand = Motor(Port.F)
force = ForceSensor(Port.C)
color = ColorSensor(Port.D)
distance = UltrasonicSensor(Port.E)
drive = DriveBase(motorl, motorr, 55, 112)
drive.settings(straight_acceleration=750)
hub.system.set_stop_button([Button.RIGHT, Button.LEFT])
hub.speaker.volume(10)
hub.speaker.play_notes(["C4/8", "C4/8", "G4/8", "G4/8"])
hub.display.icon(Icon.HAPPY)

#while True:
    #print(distance.distance(), color.color())


while not Button.CENTER in hub.buttons.pressed(): #Attend que le bouton Centre soit appuyé
    print("Appuie sur le bouton pour commencer sale toast mouillé")
drive.drive(500, 0)
wait(600)
while not distance.distance() < 1650: #Attend que la distance soit plus petite que 1550mm
    drive.drive(500, 0)
wait(500)
while not color.reflection() < 10: #Attend que la reflection de la lumière soit plus petite que 10% (noir)
    drive.drive(80, 0)
drive.settings(50) #Met la vitesse sur 50mm/s
drive.straight(35)
drive.turn(-87)
while not distance.distance() < 350:
    drive.drive(300, 0)
while not color.color() is Color.YELLOW: #Attend que la couleur soit Jaune
    drive.drive(50, 0)
drive.stop()
drive.straight(40)
drive.straight(-10)
hand.run_angle(90, -100, then=Stop.HOLD)
drive.settings(750)
drive.straight(-80)
drive.settings(500)
drive.straight(10)
hand.run_angle(90, 100)
drive.straight(-20)
drive.turn(-80)
drive.curve(650, -85, then=Stop.NONE)
drive.straight(120)
```

**Reply by dlech** [reply]
Can you run `pybricksdev run ble test_gyro.py` without `--debug` and share the output? I don't trust my counting well enough to try to guess which is line 10 or line 17.

**Reply by dlech** [reply]
Also, which firmware version are you using?

**Reply by Xylar52** [reply]
I dont have my spike with me right now but it was something like :
`
line 17 in module
Error : extra arguments or kewords given
`
But I know line 17 is the line where I initialize drivebase so
`
drive = DriveBase(motorl, motorr, 55, 112)
`

I am using the firmware available at beta.pybricks.com

**Reply by dlech** [reply]
We just updated it Monday, so we really need the exact version number to know for sure.

**Reply by dlech** [reply]
The program is working for me in Pybricks Code with firmware v3.3.0b4

**Reply by Xylar52** [reply]
Yes mine seems to also be on v3.3.0b4. I don't know what the issue seems to be. I'll let you know tomorrow if just deleting the firmware and being sure to install the lastest release works

**Reply by laurensvalk** [reply]
When you got those errors, were you perhaps running a script you made for the experimental gyro stuff?

If your script had the positive_direction or use_gyro parameters, that would explain the ‘extra keyword arguments given’ error. Those parameters didn’t make it into the beta release. To see what did make it, just browse the documentation included with beta.pybricks.com

**Reply by Xylar52** [reply]
This code was made before the imu experimental release, and it doesn't seem to contain anything related to the gyroscope. And the other scripts that had it I removed them so all are gyro-free
